test/test.ts

```diff
@@ -799,130 +801,255 @@ describe("window.codePush", function() {
+                        var deferred = Q.defer<void>();
+                        mockUpdatePackagePath = updatePath;
+                        testMessageCallback = verifyMessages([new su.AppMessage(su.TestMessage.SYNC_STATUS, [su.TestMessage.SYNC_IN_PROGRESS]),
+                            su.TestMessage.DEVICE_READY_AFTER_UPDATE],
+                            deferred);
+                        console.log("Running project...");
+                        projectManager.runPlatform(testRunDirectory, targetPlatform, true, targetEmulator).done();
+                        return deferred.promise;
+                    })
+                    .then<void>(() => {
+                        var deferred = Q.defer<void>();
+                        testMessageCallback = verifyMessages([new su.AppMessage(su.TestMessage.SYNC_STATUS, [su.TestMessage.SYNC_IN_PROGRESS])],
+                            deferred);
+                        projectManager.restartApplication(targetPlatform, TestNamespace, testRunDirectory, targetEmulator).done();
+                        return deferred.promise;
+                    })
+                    .done(done, done);
+            });
+
+            it("window.codePush.sync.2x.update", function(done) {
+                mockResponse = { updateInfo: getMockResponse(false) };
+
+                /* create an update */
+                setupUpdateProject(UpdateSync2x, "Update 1 (good update)")
+                    .then<string>(projectManager.createUpdateArchive.bind(undefined, updatesDirectory, targetPlatform))
+                    .then<void>((updatePath: string) => {
+                        var deferred = Q.defer<void>();
+                        mockUpdatePackagePath = updatePath;
+                        testMessageCallback = verifyMessages([new su.AppMessage(su.TestMessage.SYNC_STATUS, [su.TestMessage.SYNC_IN_PROGRESS])],
+                            deferred);
+                        console.log("Running project...");
+                        projectManager.runPlatform(testRunDirectory, targetPlatform, true, targetEmulator).done();
+                        return deferred.promise;
+                    })
+                    .then<void>(() => {
+                        var deferred = Q.defer<void>();
+                        var noUpdateReponse = createDefaultResponse();
+                        noUpdateReponse.isAvailable = false;
+                        noUpdateReponse.appVersion = "0.0.1";
+                        mockResponse = { updateInfo: noUpdateReponse };
+                        testMessageCallback = verifyMessages([new su.AppMessage(su.TestMessage.SYNC_STATUS, [su.TestMessage.SYNC_IN_PROGRESS])],
+                            deferred);
+                        projectManager.restartApplication(targetPlatform, TestNamespace, testRunDirectory, targetEmulator).done();
+                        return deferred.promise;
+                    })
+                    .done(done, done);
+            });
```

Comment:
I'm a bit confused as to what the goal of these tests are. It seems all of them are really only testing the same thing (that the second sync returns SYNC_IN_PROGRESS). Even though different scenarios are being run, we are not verifying the output of the first sync to verify that they are succeeding.

I would recommend either just having one test to test `IN_PROGRESS` specifically, or modifying the existing test to check for the expected output of the first `sync()` call as well.